bftengine/include/bftengine/IStateTransfer.hpp

```diff
@@ -89,6 +89,8 @@ class IStateTransfer : public IReservedPages {
   // Accepts the checkpoint number as a parameter.
   // Callbacks must not throw.
   // Multiple callbacks can be added.
+  // TODO: the callbacks should be stored in a queue/priority queue and run in a
+  // predetermined order to prevent repeating logic such as loading main keys from reserved pages
```

Comment:
If you load the callback only once, how would it executed more than once?

Reply:
I repeated the logic which parses the reserved pages because I needed it to happen prior to the async CRE fetching the state. This could have been avoided had I the option to guarantee the execution order of the callbacks.